site/app/models/notebook/Notebook.php

```diff
@@ -155,28 +155,37 @@ private function getMarkdownData($cell) {
     * @param int $version which version to get notebook submission values from
     * @param string $student_id which student's notebook to pull data from
     */
-    public function getMostRecentNotebookSubmissions(int $version, array $new_notebook, string $student_id): array {
+    public function getMostRecentNotebookSubmissions(int $version, array $new_notebook, string $student_id, int $display_version, string $gradeable_id, string $semester, string $course): array {
         foreach ($new_notebook as $notebookKey => $notebookVal) {
             if (isset($notebookVal['type'])) {
                 if ($notebookVal['type'] == "short_answer") {
                     // If no previous submissions set string to default initial_value
                     if ($version === 0) {
                         $recentSubmission = $notebookVal['initial_value'] ?? "";
+                        $version_answer = $notebookVal['initial_value'] ?? "";
                     }
                     else {
                         // Else there has been a previous submission try to get it
                         try {
                             // Try to get the most recent submission
                             $recentSubmission = $this->getRecentSubmissionContents($notebookVal['filename'], $version, $student_id);
+                            //get answer for the selected display version
+                            $question_name = $notebookVal['filename'];
+                            $file = fopen("/var/local/submitty/courses/" . $semester . "/" . $course . "/submissions/" . $gradeable_id . "/" . $student_id . "/" . $display_version . "/" . $question_name, "r");
+                            $version_answer = fread($file, filesize("/var/local/submitty/courses/" . $semester . "/" . $course . "/submissions/" . $gradeable_id . "/" . $student_id . "/" . $display_version . "/" . $question_name));
```

Comment:
It would be better to use `$this->core->getConfig()->getCoursePath()` here instead of hardcoded paths.  That would also mean that you wouldn't need the `$semester` and `$course` parameters I believe.